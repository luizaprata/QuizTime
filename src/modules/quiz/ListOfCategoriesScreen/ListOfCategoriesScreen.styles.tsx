import colors from '@/resources/colors';
import styled from 'styled-components/native';

export const ButtonContainer = styled.View`
  padding: 50px;
  background-color: ${colors.chumbo200};
`;
