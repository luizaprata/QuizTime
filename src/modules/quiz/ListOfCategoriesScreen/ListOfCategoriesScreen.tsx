import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { ButtonContainer } from './ListOfCategoriesScreen.styles';
import useAllCategoriesApi from './useAllCategoriesApi';
import { Button, Text } from 'react-native';
import { Category } from '@/modules/quiz/types/Trivia.types';
import { ScreenArea, ScrollArea } from '@/components/Screen/Screen.styles';
import { AppScreensEnum } from '@/types/AppScreensEnum';

const ListOfCategoriesScreen: React.FC = () => {
  const navigation = useNavigation();

  const { payload, isLoading, errorMessage } = useAllCategoriesApi();

  const onCategorySelected = (category: Category) => {
    navigation.navigate(AppScreensEnum.QuizByCategory, {
      category,
    });
  };

  if (errorMessage) {
    return <Text>{errorMessage}</Text>;
  }
  return (
    <ScreenArea>
      <ScrollArea>
        {isLoading && <Text>Carregando</Text>}
        {!isLoading && (
          <ButtonContainer>
            {payload?.trivia_categories?.map((category) => {
              return (
                <Button
                  key={category.id}
                  title={category.name}
                  onPress={() => onCategorySelected(category)}
                />
              );
            })}
          </ButtonContainer>
        )}
      </ScrollArea>
    </ScreenArea>
  );
};

export default ListOfCategoriesScreen;
