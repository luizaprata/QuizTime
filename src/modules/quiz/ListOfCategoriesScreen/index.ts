import ListOfCategoriesScreen from './ListOfCategoriesScreen';

export default ListOfCategoriesScreen;
