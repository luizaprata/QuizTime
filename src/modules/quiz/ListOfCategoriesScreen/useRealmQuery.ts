import { useMemo, useContext } from 'react';
import DatabaseContext from '../infrastructure/database/DatabaseContext';

export interface IUseRealmQueryParams<T> {
  source: string | Realm.Results<T>;
  isFocused: boolean;
}

export default function useRealmQuery<T>({
  source,
  isFocused,
}: IUseRealmQueryParams<T>): Realm.Collection<T> | undefined {
  const { realm } = useContext(DatabaseContext);
  const query = useMemo(() => {
    return realm && isFocused && typeof source === 'string'
      ? realm.objects<T>(source)
      : undefined;
  }, [realm, source, isFocused]);

  return query;
}
