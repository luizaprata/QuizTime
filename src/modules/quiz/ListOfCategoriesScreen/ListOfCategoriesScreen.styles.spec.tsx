import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonContainer } from './ListOfCategoriesScreen.styles';

describe('ListOfCategoriesScreen.styles', () => {
  test('THEN Container SHOULD match with snapshot', () => {
    const tree = renderer.create(<ButtonContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
