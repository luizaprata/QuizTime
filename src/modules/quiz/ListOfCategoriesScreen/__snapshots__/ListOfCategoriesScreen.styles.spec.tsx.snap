// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfCategoriesScreen.styles THEN Container SHOULD match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F4F4F4",
        "paddingBottom": 50,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 50,
      },
    ]
  }
/>
`;
